.github/scripts/trymerge.py

```
@@ -699,24 +699,34 @@ def merge_ghstack_into(self, repo: GitRepo, force: bool, comment_id: Optional[in
-            repo._run_git("commit", f"--author=\"{self.get_author()}\"", "-m", msg)
-        else:
-            self.merge_ghstack_into(repo, force, comment_id=comment_id)
-
-        repo.push(self.default_branch(), dry_run)
-        if not dry_run:
-            gh_add_labels(self.org, self.project, self.pr_num, ["merged"])
+
+        # Adds a dumb retry for now as we scope a longer term soln to https://github.com/pytorch/pytorch/issues/78176
+        # Should catch any exceptions where the master branch is newer than the last pull
+        for _ in range(5):
+            try:
+                if repo.current_branch() != self.default_branch():
+                    repo.checkout(self.default_branch())
+                if not self.is_ghstack_pr():
+                    # Adding the url here makes it clickable within the Github UI
+                    approved_by_urls = ', '.join(prefix_with_github_url(login) for login in self.get_approved_by())
+                    msg = self.get_title() + f" (#{self.pr_num})\n\n" + self.get_body()
+                    msg += f"\nPull Request resolved: {self.get_pr_url()}\n"
+                    msg += f"Approved by: {approved_by_urls}\n"
+                    pr_branch_name = f"__pull-request-{self.pr_num}__init__"
+                    repo.fetch(f"pull/{self.pr_num}/head", pr_branch_name)
+                    repo._run_git("merge", "--squash", pr_branch_name)
+                    repo._run_git("commit", f"--author=\"{self.get_author()}\"", "-m", msg)
+                else:
+                    self.merge_ghstack_into(repo, force, comment_id=comment_id)
+
+                repo.push(self.default_branch(), dry_run)
+                if not dry_run:
+                    gh_add_labels(self.org, self.project, self.pr_num, ["merged"])
+
+                return
+            except Exception as e:
+                print(f"Could not merge {self.pr_num} because: {e}.")
+        print(f"All retries exhausted, merging {self.pr_num} failed.")
```

This approach eats the exception instead of letting it propagate up, which I think will mess up our failure reporting.

---

Oh true. We probably want to split up the merge_into function.